tabpy-client/tabpy_client/client.py

```diff
@@ -31,15 +31,28 @@
     unicode = str
 
 
-def _check_endpoint_name(name):
```

Comment:
There are two places that call _check_endpoint_name(): alias() at line 246 and _gen_endpoint() at line 385. It seems like you're replacing that function with two separate functions. Shouldn't you update the callers of the function to also exercise _check_hostname()?

Reply:
Changing those other two lines could break the ability to deploy code from the client to the server. The term endpoint is a bit confusing here, as it is used in a couple of different ways, and I did not try and change any of the existing wording to make that distinction more clear. 

Those other two calls are for checking the name of the endpoint which is being published to TabPy Server (e.g. client.deploy('add', add, 'Adds two numbers x and y') ). 

The addition of the _check_hostname function, is used to provide a check to the hostname that is used when instantiating the Client (e.g.  client = tabpy_client.Client('http://localhost:9004/') ) and will be a more useful check when bringing in remote deploy functionality.

I'm up for reducing mental complexity, so would be willing to make those additional changes to make the distinction clear. I don't expect there would be too many places that change would need to be made, but at least in the Client() function and the unittest.

Reply:
Thanks for the helpful comment! I agree that refactoring to reduce mental complexity might be helpful. That said, no reason to have that gate this CR.

LGTM!